### DISCUSSION TITLE
Enable lazy evaluation in iris.analysis.cartography.rotate_winds

### DISCUSSION THREAD

**Discussion by tinyendian** [SciTools/iris#4934] [General]
Dear Iris developers,

I am currently working on enabling out-of-core computation in a file format conversion tool that uses Iris to load CF-netCDF data. Everything is working fine, including Iris regridding, but I noticed that the `rotate_winds` function in the cartography module is not currently lazy evaluation-enabled, so it will force wind field cube data to be loaded.

With a few straightforward modifications, I was able to enable lazy evaluation, including array masking. So I was wondering if there is any interest in this? I'd be happy to contribute an issue and pull request with the necessary changes. There seem to be more functions in the cartography module without lazy evaluation, though.

**Comment by pp-mo** [comment]
Not sure if this was somehow missed (we should have spotted it in our "Peloton" meetings).
Anyway, we would certainly welcome a PR for it, especially if it's fairly straightforward as you suggest.

> There seem to be more functions in the cartography module without lazy evaluation, though.

By all means include other functions if they are interest to you, but I wouldn't worry about completeness.
These changes aren't a strategic goal right now, so incremental improvement is fine.

**Reply by tinyendian** [reply]
Hi @pp-mo, no worries, and thanks for your reply - I'll create an issue for this, and will be back with a pull request soon.
